core/src/main/scala/org/broadinstitute/dsde/rawls/billing/GoogleBillingProjectLifecycle.scala

```diff
@@ -54,10 +66,19 @@ class GoogleBillingProjectLifecycle(samDAO: SamDAO, gcsDAO: GoogleServicesDAO)(i
       _ <- syncBillingProjectOwnerPolicyToGoogleAndGetEmail(samDAO, createProjectRequest.projectName)
     } yield CreationStatuses.Ready
 
-  override def preDeletionSteps(projectName: RawlsBillingProjectName, ctx: RawlsRequestContext): Future[Unit] =
+  override def initiateDelete(projectName: RawlsBillingProjectName, ctx: RawlsRequestContext)(implicit
+    executionContext: ExecutionContext
+  ): Future[(UUID, JobType)] =
     // Note: GoogleBillingProjectLifecycleSpec does not test that this method is called because the method
     // lives in a companion object (which makes straight mocking impossible), and the method will be removed
     // once workspace migration is complete. Note also that the more "integration" level test BillingApiServiceV2Spec
     // does verify that code in this method is executed when a Google-based project is deleted.
     deleteGoogleProjectIfChild(projectName, ctx.userInfo, gcsDAO, samDAO, ctx)
+      .map(_ => (UUID.randomUUID(), GoogleBillingProjectDelete))
+
+  override def finalizeDelete(projectName: RawlsBillingProjectName, ctx: RawlsRequestContext)(implicit
+    executionContext: ExecutionContext
+  ): Future[Unit] =
+    unregisterBillingProject(projectName, ctx)
```

Comment:
This is just what the orchestrator was doing after the `preDeletionSteps` was finished